diff --git a/buf.c b/buf.c
index d8992f7..1ae7542 100644
--- a/buf.c
+++ b/buf.c
@@ -1261,7 +1261,7 @@ xmlBufMergeBuffer(xmlBufPtr buf, xmlBufferPtr buffer) {
  */
 int
 xmlBufResetInput(xmlBufPtr buf, xmlParserInputPtr input) {
-    if ((input == NULL) || (buf == NULL) || (buf->error))
+    if ((input == NULL) || (buf == NULL))
         return(-1);
     CHECK_COMPAT(buf)
     input->base = input->cur = buf->content;
